apps/issuance/src/issuance.service.ts

```diff
@@ -808,16 +803,45 @@ async sendEmailForCredentialOffer(sendEmailCredentialOffer: SendEmailCredentialO
     }
   }
 
-  async exportSchemaToCSV(credentialDefinitionId: string): Promise<object> {
+  async exportSchemaToCSV(templateDetails: TemplateDetailsInterface): Promise<object> {
     try {
-      const schemaResponse: SchemaDetails = await this.issuanceRepository.getCredentialDefinitionDetails(credentialDefinitionId);
+      let schemaResponse: SchemaDetails;
+      let fileName: string;
+
+      const {schemaType, templateId} = templateDetails;
 
+      if (!templateId) {
+        throw new BadRequestException(ResponseMessages.bulkIssuance.error.invalidtemplateId);
+      }
+      const timestamp = Math.floor(Date.now() / 1000);
+      
+      if (schemaType === SchemaType.INDY) {
+         schemaResponse = await this.issuanceRepository.getCredentialDefinitionDetails(templateId);
+
+         if (!schemaResponse) {
+          throw new NotFoundException(ResponseMessages.bulkIssuance.error.invalidIdentifier);
+        }
+
+         fileName = `${schemaResponse.tag}-${timestamp}.csv`;
+
+      } else if (schemaType === SchemaType.W3C_Schema) {
+        const schemDetails = await this.issuanceRepository.getSchemaDetailsBySchemaIdentifier(templateId);
+        const {attributes, schemaLedgerId, name} = schemDetails;
+        schemaResponse.attributes = attributes;
+        schemaResponse.schemaLedgerId = schemaLedgerId;
+        schemaResponse.name = name;
+
+        if (!schemaResponse) {
```

Comment:
Just a thought, can we use this validation for both? We've already considered schemaResponse as a let, so can we use this block of code outside this condition?

Reply:
After the validation, I assigned file name and the file name will be different for both the schema type